### PR TITLE
Fix ellipsoidal area calculation

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

**Comment by nyalldawson** [comment]
Given the nature of this, I think an in-depth review would be welcome!

@wonder-sk ?

**Comment by m-kuhn** [comment]
Are the expected test results confirmed?
Does it give better results (equal to grass) for the areas under discussion on the mailing list?
If yes, I think definitely needs merging.

**Review by wonder-sk — APPROVED** [review]
Well spotted @nyalldawson - wikipedia confirms that eccentricity^2 = 1 - (b^2 / a^2)  ... and not the other way around

File: tests/src/core/testqgsdistancearea.cpp [hunk 1]
```diff
@@ -350,13 +350,13 @@ void TestQgsDistanceArea::measureAreaAndUnits()
   area = da.measureArea( polygon );
   units = da.areaUnits();
   QgsDebugMsg( QString( "measured %1 in %2" ).arg( area ).arg( QgsUnitTypes::toString( units ) ) );
-  QGSCOMPARENEAR( area, 184149.37, 1.0 );
+  QGSCOMPARENEAR( area, 185818.590966, 1.0 );
```
Comment (nyalldawson) [inline tests/src/core/testqgsdistancearea.cpp]: Arcgis: 185825

File: tests/src/core/testqgsdistancearea.cpp [hunk 2]
```diff
@@ -350,13 +350,13 @@ void TestQgsDistanceArea::measureAreaAndUnits()
...
   QCOMPARE( units, QgsUnitTypes::AreaSquareMeters );
 
   // test converting the resultant area
   area = da.convertAreaMeasurement( area, QgsUnitTypes::AreaSquareYards );
   QgsDebugMsg( QString( "measured %1 in sq yrds" ).arg( area ) );
-  QGSCOMPARENEAR( area, 220240.8172549, 0.3 );
+  QGSCOMPARENEAR( area, 222237.185213, 0.3 );
```
Comment (nyalldawson) [inline tests/src/core/testqgsdistancearea.cpp]: Arcgis: 222244

File: tests/src/python/test_qgsdistancearea.py [hunk 3]
```diff
@@ -551,12 +551,12 @@ def testAreaMeasureAndUnits(self):
 
         print(("measured {} in {}".format(area, QgsUnitTypes.toString(units))))
         # should always be in Meters Squared
-        self.assertAlmostEqual(area, 37416879192.9, delta=0.1)
+        self.assertAlmostEqual(area, 36918093794.121284, delta=0.1)
```
Comment (nyalldawson) [inline tests/src/python/test_qgsdistancearea.py]: Arcgis: 36922805922

File: tests/src/python/test_qgsdistancearea.py [hunk 4]
```diff
@@ -551,12 +551,12 @@ def testAreaMeasureAndUnits(self):
...
         self.assertEqual(units, QgsUnitTypes.AreaSquareMeters)
 
         # test converting the resultant area
         area = da.convertAreaMeasurement(area, QgsUnitTypes.AreaSquareMiles)
-        self.assertAlmostEqual(area, 14446.7378, delta=0.001)
+        self.assertAlmostEqual(area, 14254.155703182701, delta=0.001)
```
Comment (nyalldawson) [inline tests/src/python/test_qgsdistancearea.py]: Arcgis: 14255.975065928

File: tests/src/python/test_qgsdistancearea.py [hunk 5]
```diff
@@ -582,12 +582,12 @@ def testAreaMeasureAndUnits(self):
         area = da.measureArea(polygon)
         units = da.areaUnits()
         print(("measured {} in {}".format(area, QgsUnitTypes.toString(units))))
-        self.assertAlmostEqual(area, 184149.37, delta=1.0)
+        self.assertAlmostEqual(area, 185818.59096575077, delta=1.0)
```
Comment (nyalldawson) [inline tests/src/python/test_qgsdistancearea.py]: Arcgis: 185825

File: tests/src/python/test_qgsdistancearea.py [hunk 6]
```diff
@@ -582,12 +582,12 @@ def testAreaMeasureAndUnits(self):
...
         self.assertEqual(units, QgsUnitTypes.AreaSquareMeters)
 
         # test converting the resultant area
         area = da.convertAreaMeasurement(area, QgsUnitTypes.AreaSquareYards)
-        self.assertAlmostEqual(area, 220240.8172549, delta=1.0)
+        self.assertAlmostEqual(area, 222237.18521272976, delta=1.0)
```
Comment (nyalldawson) [inline tests/src/python/test_qgsdistancearea.py]: Arcgis: 222244

**Pull request merged**